### PR TITLE
Add abandoned colony and fix bugs with loading

### LINKED ISSUE #1299: Multiple Colonies in SSP
Had a few requests about adding the ability in SSP for allowing multiple colonies in the same world.

Is there a technical reason we can't do this, or a massive amount of work involved in adding this in as a config option?

Thanks,

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #1299 

# Changes proposed in this pull request:
- This fixes a bug where not enough colonies get loaded after deleting more than the buffer.
- This allows to abandon your colony now. Just transfer ownership to [abandoned] and the colony will be transferred to a fake player allowing the player to create a new colony. He can also add himself as an officer via command so he might even control both.
- Fix bug where sleeping colonists die (since they're not in a bed and search a tp location)

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java [hunk 1]
```diff
@@ -50,7 +53,6 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
     @Override
     public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java:54]: MAJOR Reduce the number of returns of this method 9, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
MAJOR The Cyclomatic Complexity of this method "execute" is 16 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
MAJOR Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 61 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 57 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ColonyManager.java#L764](https://github.com/Minecolonies/minecolonies/blob/efee5661a622a0c2835860b6f436bfbefd221d47/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L764): Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/efee5661a622a0c2835860b6f436bfbefd221d47/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/efee5661a622a0c2835860b6f436bfbefd221d47/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/efee5661a622a0c2835860b6f436bfbefd221d47/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/efee5661a622a0c2835860b6f436bfbefd221d47/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [ColonyManager.java#L49](https://github.com/Minecolonies/minecolonies/blob/efee5661a622a0c2835860b6f436bfbefd221d47/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L49): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 35 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [ColonyManager.java#L49](https://github.com/Minecolonies/minecolonies/blob/efee5661a622a0c2835860b6f436bfbefd221d47/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L49): class "ColonyManager" has 53 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ColonyManager.java#L206](https://github.com/Minecolonies/minecolonies/blob/efee5661a622a0c2835860b6f436bfbefd221d47/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L206): Provide multiple methods instead of using "canDestroy" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [ColonyManager.java#L736](https://github.com/Minecolonies/minecolonies/blob/efee5661a622a0c2835860b6f436bfbefd221d47/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L736): Refactor this method to reduce its Cognitive Complexity from 31 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [ColonyManager.java#L823](https://github.com/Minecolonies/minecolonies/blob/efee5661a622a0c2835860b6f436bfbefd221d47/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L823): Either log or rethrow this exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 99 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 84 major
* ![MINOR][MINOR] 13 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ColonyManager.java#L764](https://github.com/Minecolonies/minecolonies/blob/672ca380d8d1e5420e794fb6ade8169b93cce0d9/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L764): Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/672ca380d8d1e5420e794fb6ade8169b93cce0d9/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/672ca380d8d1e5420e794fb6ade8169b93cce0d9/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/672ca380d8d1e5420e794fb6ade8169b93cce0d9/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/672ca380d8d1e5420e794fb6ade8169b93cce0d9/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [ColonyManager.java#L49](https://github.com/Minecolonies/minecolonies/blob/672ca380d8d1e5420e794fb6ade8169b93cce0d9/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L49): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 35 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [ColonyManager.java#L49](https://github.com/Minecolonies/minecolonies/blob/672ca380d8d1e5420e794fb6ade8169b93cce0d9/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L49): class "ColonyManager" has 53 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ColonyManager.java#L206](https://github.com/Minecolonies/minecolonies/blob/672ca380d8d1e5420e794fb6ade8169b93cce0d9/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L206): Provide multiple methods instead of using "canDestroy" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [ColonyManager.java#L736](https://github.com/Minecolonies/minecolonies/blob/672ca380d8d1e5420e794fb6ade8169b93cce0d9/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L736): Refactor this method to reduce its Cognitive Complexity from 31 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [ColonyManager.java#L823](https://github.com/Minecolonies/minecolonies/blob/672ca380d8d1e5420e794fb6ade8169b93cce0d9/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L823): Either log or rethrow this exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 102 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 86 major
* ![MINOR][MINOR] 14 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ColonyManager.java#L764](https://github.com/Minecolonies/minecolonies/blob/ebb1dd6d7b89e5cd02ddb98ab2a93199839928f1/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L764): Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/ebb1dd6d7b89e5cd02ddb98ab2a93199839928f1/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/ebb1dd6d7b89e5cd02ddb98ab2a93199839928f1/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/ebb1dd6d7b89e5cd02ddb98ab2a93199839928f1/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/ebb1dd6d7b89e5cd02ddb98ab2a93199839928f1/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [ColonyManager.java#L49](https://github.com/Minecolonies/minecolonies/blob/ebb1dd6d7b89e5cd02ddb98ab2a93199839928f1/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L49): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 35 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [ColonyManager.java#L49](https://github.com/Minecolonies/minecolonies/blob/ebb1dd6d7b89e5cd02ddb98ab2a93199839928f1/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L49): class "ColonyManager" has 53 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ColonyManager.java#L206](https://github.com/Minecolonies/minecolonies/blob/ebb1dd6d7b89e5cd02ddb98ab2a93199839928f1/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L206): Provide multiple methods instead of using "canDestroy" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [ColonyManager.java#L736](https://github.com/Minecolonies/minecolonies/blob/ebb1dd6d7b89e5cd02ddb98ab2a93199839928f1/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L736): Refactor this method to reduce its Cognitive Complexity from 31 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [ColonyManager.java#L823](https://github.com/Minecolonies/minecolonies/blob/ebb1dd6d7b89e5cd02ddb98ab2a93199839928f1/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L823): Either log or rethrow this exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Pull request merged**